metecho/api/jobs.py

```diff
@@ -146,6 +155,135 @@ def _create_branches_on_github(
     return task_branch_name
 
 
+def create_repository(
+    project: Project,
+    *,
+    user: User,
+    dependencies: Iterable[str],
+    template_repo_owner: str = None,
+    template_repo_name: str = None,
+):
+    """
+    Given a local Metecho Project create and bootstrap the corresponding GitHub repository.
+    """
+    project.refresh_from_db()
+    repo = None
+
+    try:
+        # Ensure the user is part of the org that owns the project
+        user_gh = gh_as_user(user)
+        user_orgs = [org.login for org in user_gh.organizations()]
+        if project.repo_owner not in user_orgs:
+            raise ValueError(
+                _(
+                    "Either you are not a member of the %(name)s organization "
+                    "or it hasn't installed the Metecho GitHub app"
+                )
+                % {"name": project.repo_owner}
+            )
+
+        # Get a GitHub session with write permissions on the org
+        org_gh = gh_as_org(project.repo_owner)
+        org = org_gh.organization(project.repo_owner)
+        if template_repo_owner and template_repo_name:
+            tpl_repo = org_gh.repository(template_repo_owner, template_repo_name)
+            branch_name = tpl_repo.default_branch or "main"
+        else:
+            tpl_repo = None
+            branch_name = "main"
+
+        # Create team & repository on GitHub
+        team = org.create_team(f"{project} Team")
```

Comment:
This raises a 422 error if the team name is unavailable. Maybe we should iterate and append numbers or something?

```
Traceback (most recent call last):
  File "/app/metecho/api/jobs.py", line 196, in create_repository
    team = org.create_team(f"{project} Team")
  File "/usr/local/lib/python3.10/site-packages/github3/decorators.py", line 24, in auth_wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/github3/orgs.py", line 761, in create_team
    json = self._json(self._post(url, data), 201)
  File "/usr/local/lib/python3.10/site-packages/github3/models.py", line 161, in _json
    raise exceptions.error_for(response)
github3.exceptions.UnprocessableEntity: 422 Validation Failed

id=no-job-id at=ERROR time="2022-05-23 19:25:20.000010" tag=external module=worker msg="Traceback (most recent call last):
  File \"/usr/local/lib/python3.10/site-packages/rq/worker.py\", line 1061, in perform_job
    rv = job.perform()
  File \"/usr/local/lib/python3.10/site-packages/rq/job.py\", line 821, in perform
    self._result = self._execute()
  File \"/usr/local/lib/python3.10/site-packages/rq/job.py\", line 844, in _execute
    result = self.func(*self.args, **self.kwargs)
  File \"/app/metecho/api/jobs.py\", line 196, in create_repository
    team = org.create_team(f\"{project} Team\")
  File \"/usr/local/lib/python3.10/site-packages/github3/decorators.py\", line 24, in auth_wrapper
    return func(self, *args, **kwargs)
  File \"/usr/local/lib/python3.10/site-packages/github3/orgs.py\", line 761, in create_team
    json = self._json(self._post(url, data), 201)
  File \"/usr/local/lib/python3.10/site-packages/github3/models.py\", line 161, in _json
    raise exceptions.error_for(response)
github3.exceptions.UnprocessableEntity: 422 Validation Failed
```

Reply:
Fixed in 8e1d2481d85118bd7a87218de662ab40711f9d9f